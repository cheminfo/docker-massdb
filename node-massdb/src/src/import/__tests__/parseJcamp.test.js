'use strict';

let join = require('path').join;

let parseJcamp = require('../parseJcamp');

test('creation of jcamp parsing', async () => {
  let file = join(__dirname, 'data/nist.jdx');
  let result = await parseJcamp(file);
  expect(Object.keys(result)).toHaveLength(2);
  expect(result).toMatchSnapshot();
});
