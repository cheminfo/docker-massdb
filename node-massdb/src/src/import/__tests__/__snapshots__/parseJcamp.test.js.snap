// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creation of jcamp parsing 1`] = `
Array [
  Spectrum {
    "data": Array [
      Object {
        "x": Array [
          1,
          2,
        ],
        "y": Array [
          21,
          999,
        ],
      },
    ],
    "dataType": "MASS SPECTRUM",
    "isPeaktable": true,
    "name": "Hydrogen",
    "nbPoints": 2,
    "title": "Hydrogen",
    "xFactor": 1,
    "yFactor": 1,
  },
  Spectrum {
    "data": Array [
      Object {
        "x": Array [
          2,
          4,
        ],
        "y": Array [
          15,
          999,
        ],
      },
    ],
    "dataType": "MASS SPECTRUM",
    "isPeaktable": true,
    "name": "Deuterium",
    "nbPoints": 2,
    "title": "Deuterium",
    "xFactor": 1,
    "yFactor": 1,
  },
]
`;
